src/main/java/org/broadinstitute/hellbender/tools/walkers/PairWalker.java

```diff
@@ -84,21 +88,42 @@ public void apply( final GATKRead read,
             applyUnpaired(read);
             return;
         }
+
+        final String contig = read.getAssignedContig();
+        if ( contig != null &&
+                !contig.equals(ReadConstants.UNSET_CONTIG) &&
+                !contig.equals(curContig) ) {
+            clearBufferSet();
+            curContig = contig;
+        }
+
         final boolean inInterval = regionChecker == null || regionChecker.isInInterval(read);
         final PairBufferEntry pb = new PairBufferEntry(read, inInterval);
         final PairBufferEntry pbMate = pairBufferSet.find(pb);
         if ( pbMate == null ) {
-            pairBufferSet.add(pb);
-        } else {
+            // If it's in one of our intervals, it's always interesting -- hopefully we'll find the mate.
+            // And if it's the first to arrive of the pair, it just might be interesting, anyway.
+            // (Actually, if it's not definitely the 2nd of the pair, we'll save it. There can be doubt.)
+            // But if it's not in the interval and its mate is upstream, we'd expect the mate to be in
+            //   the buffer set already.  The fact that it's not means that the mate must not have been
+            //   near any of the intervals: so the pair is uninteresting, and we don't add it to the
+            //   buffer set.
+            if ( inInterval || !pb.isDownstreamMate() ) {
+                pairBufferSet.add(pb);
+            }
+        } else if ( pb.getRead().isFirstOfPair() != pbMate.getRead().isFirstOfPair() ) {
+            // above check is necessary because under rare circumstances you might see the
+            //  real read and it's distant-mate shadow before seeing the other read in the pair
```

Comment:
Is there a unit test that hits this case? If there's not it might be helpful to add one, if there is one and I'm not realizing it maybe a comment in the unit test class might help point it out.